### ISSUE TITLE
[Prims+NvFuser] Non-fusible ops Tracker

### ISSUE THREAD

**Issue by SherlockNoMad** [pytorch/pytorch#81307]
Following is the complete list of ops that appear in the torch bench models, but are not included in the nvFuser fusion group.

They are not included due to following reasons: 
1. nvFuser backend does not support this op
2. aten2aten missing 
3. aten2prim ref missing 
4. nvfuser_impl missing for prims op

- [ ] 'aten._to_copy.default'
- [ ] 'aten.mean.dim'
- [ ] 'aten.select.int'
- [ ] 'aten.upsample_bilinear2d.vec'
- [ ] 'aten._unsafe_view.default'
- [ ] 'aten.avg_pool2d.default'
- [ ] 'aten.empty_like.default'
- [ ] 'aten.bmm.default'
- [ ] 'aten.embedding_dense_backward.default'
- [ ] 'aten.where.ScalarSelf'
- [ ] 'aten.scatter_add.default'
- [ ] 'aten.zero.default'
- [ ] 'aten.where.self'
- [ ] 'aten.to.dtype'
- [ ] 'aten.max_pool2d_with_indices.default'
- [ ] 'aten._adaptive_avg_pool2d.default'
- [ ] 'aten._embedding_bag.default'
- [ ] 'aten.clone.default'
- [ ] 'aten.expand.default'
- [ ] 'aten.slice.Tensor'
- [ ] 'aten.upsample_bilinear2d_backward.vec'
- [ ] 'aten.var.dim'
- [ ] 'aten.permute.default'
- [ ] 'aten.gather.default'
- [ ] 'aten.square.default'
- [ ] 'aten.leaky_relu.default'
- [ ] 'aten.fill.Scalar'
- [ ] 'aten._adaptive_avg_pool2d_backward.default'
- [ ] 'aten.slice_scatter.default'
- [ ] 'aten.index.Tensor'
- [ ] 'aten.view.default'
- [ ] 'aten.squeeze.dim'
- [ ] 'aten.index_select.default'
- [ ] 'aten.reflection_pad2d_backward.default'
- [ ] 'aten.avg_pool2d_backward.default'
- [ ] 'aten.repeat.default'
- [ ] 'aten.max_pool2d_with_indices_backward.default'
- [ ] 'aten.reflection_pad2d.default'
- [ ] 'aten.hardtanh.default'
- [ ] 'aten.unsqueeze.default'
- [ ] 'aten.t.default'
- [ ] 'aten.grid_sampler_2d.default'
- [ ] 'aten.constant_pad_nd.default'
- [ ] 'aten.new_zeros.default'
- [ ] 'aten.mean.default'
- [ ] 'aten.mm.default'
- [ ] 'aten.native_layer_norm_backward.default'
- [ ] 'aten.narrow.default'
- [ ] 'aten.convolution_backward.default'
- [ ] 'aten.new_empty.default'
- [ ] 'aten.stack.default'
- [ ] 'aten.select_scatter.default'
- [ ] 'aten.amax.default'
- [ ] 'aten.convolution.default'
- [ ] 'aten.addmm.default'
- [ ] 'aten.cat.default'

**Comment by gchanan** [comment]
Can we do the same exercise for HF-Transformer and TIMM models?

**Comment by SherlockNoMad** [comment]
> Can we do the same exercise for HF-Transformer and TIMM models?

Yes definitely!

**Comment by SherlockNoMad** [comment]
> @Chillee, @SherlockNoMad, why does `where.ScalarSelf` appear here? It should redispatch to `where.self`.

This is coming from aten.relu's ref. 

```
@register_decomposition(torch.ops.aten.relu)
@elementwise_type_promotion_wrapper(
    type_promoting_args=("a",),
    type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.DEFAULT,
)
def relu(a: TensorLikeType, inplace: bool = False) -> TensorLikeType:
    """
    Reference implementation of torch.nn.functional.relu
    """

    if inplace:
        raise NotImplementedError

    return torch.where(torch.le(a, 0), 0, a)
```

aten2prim decomp:  aten.relu.default <function relu at 0x7fa91af114c0>
aten2prim decomp:  aten.le.Scalar <function _make_elementwise_binary_reference.<locals>._ref at 0x7fa91af43940>
aten2prim decomp:  aten.where.ScalarSelf <function where at 0x7fa91af65dc0>

**Comment by ngimel** [comment]
ugh, it should further be decomposed to fill_ and `aten.where.self` (like happens naturally with aot tracing), multi-stage decompositions are inconvenient. Or maybe nvfuser can support where.ScalarSelf, I think the infra is all in place, and it should just work.

**Comment by ngimel** [comment]
Yeah, I meant python APIs should add the necessary overload and hopefully it will just work? cc @jjsjann123, @kevinstephano

**Comment by kevinstephano** [comment]
The `prim` for `where` is hooked up here: https://github.com/pytorch/pytorch/blob/80f6d2e9e66a0d133a24863e6338f4c3a493e2fc/torch/_prims/__init__.py#L1935-L1942

Is there some other connection that needs to be made? It's not clear to me how `aten.where.ScaleSelf` gets translated to the `prim` `where.

**Comment by ngimel** [comment]
Hm, right, it looks like it already has scalar overloads, so why then the error message in #81358 ?

**Comment by SherlockNoMad** [comment]
> Can we do the same exercise for HF-Transformer and TIMM models?

Include HF-Transformer models. The list didn't changed much, only 3 new ops showed up: aten.any, aten.col2im, aten.im2col

**Comment by jjsjann123** [comment]
BTW, this list should be updated, as we now only apply a handful of selected aten2aten decomposition, I'm seeing more aten ops showing up as not-fusible-ops.
